core/src/main/java/org/apache/carbondata/core/metadata/SegmentFileStore.java

```diff
@@ -488,6 +485,49 @@ private void readIndexFiles(SegmentStatus status, boolean ignoreStatus) throws I
     }
   }
 
+  /**
+   * Reads all merge index / index files as per the status of the file.
+   * In case of @ignoreStatus is true it just reads all merge index / index files
+   *
+   * @param status
+   * @param ignoreStatus
+   * @return
+   * @throws IOException
+   */
+  private List<String> readIndexOrMergeFiles(SegmentStatus status, boolean ignoreStatus)
```

Comment:
why do you need this method, Already merge files are available in `SegmentIndexFileStore.getCarbonMergeFileToIndexFilesMap`